src/common/types.cpp

```diff
@@ -1042,33 +1044,50 @@ struct EnumTypeInfo : public ExtraTypeInfo {
 			return false;
 		}
 		auto &other = (EnumTypeInfo &)*other_p;
-		// We must check if all elements are equal
-		if (other.values_insert_order == values_insert_order) {
-			return true;
+
+		// We must check if both enums have the same size
+		if (other.size != size) {
+			return false;
 		}
-		return false;
+		auto other_vector_ptr = FlatVector::GetData<string_t>(other.values_insert_order);
+		auto this_vector_ptr = FlatVector::GetData<string_t>(values_insert_order);
+
+		// Now we must check if all strings are the same
+		for (idx_t i = 0; i < size; i++) {
+			auto str_size = this_vector_ptr[i].GetSize();
+			if (other_vector_ptr[i].GetSize() != this_vector_ptr[i].GetSize()) {
```

Comment:
Can we use`Equals::Operation` here, instead of reimplementing the string equals? (`src/include/duckdb/common/operator/comparison_operators.hpp`)